File: 0179-largest-number/0179-largest-number.cpp
class Solution {
public:
    static bool compare (string& a, string& b) {
        return a+b>b+a;
    }
    
    string largestNumber(vector<int>& nums) {
        vector<string> str;
        for (auto num : nums) 
            str.push_back(to_string(num));
        sort(str.begin(), str.end(), compare);
        string result;
        for (auto num: str) 
            result += num;
        while (result[0] == '0' && result.length()>1) 
            result.erase(0,1);
        return result;
    }
};

File: 0179-largest-number/0179-largest-number.py
class Solution:
    def largestNumber(self, nums: List[int]) -> str:

        def compare(a, b):
            return str(a) + str(b) > str(b) + str(a)
        
        for i in range(len(nums), 0, -1):
            for j in range(i-1):
                if not compare(nums[j], nums[j+1]):
                    nums[j], nums[j+1] = nums[j+1], nums[j]
        return str(int("".join(map(str, nums))))